src/rviz/InteractiveMarkerViewer.cpp

```diff
@@ -49,6 +49,39 @@ FrameMarkerPtr InteractiveMarkerViewer::addFrame(
   return marker;
 }
 
+
+
+void InteractiveMarkerViewer::visualizeTSR(const aikido::constraint::TSR& _tsr,
+  int nSamples)
+{
+  using dart::dynamics::Frame;
+  using dart::dynamics::SimpleFrame;
+  using aikido::constraint::TSR;
+
+  auto sampler = _tsr.createSampleGenerator();
+  auto state = _tsr.getSE3()->createState();
+
+  for(int i = 0; i < nSamples && sampler->canSample(); ++i)
+  {
+    sampler->sample(state);
+
+    std::stringstream sstm;
+    sstm << "tsr" << i;
+
+    auto tsrFrame = std::make_shared<SimpleFrame>(
+      Frame::World(), sstm.str(), state.getIsometry());
+    mSimpleFrames.push_back(tsrFrame);
+    this->addFrame(tsrFrame.get());
```

Comment:
Nit: We don't typically explicitly specify `this->` unless there is a name conflict.

Also, I'd prefer to aggregate all of the `FrameMarkerPtr`s returned by `addFrame` into an RAII object (perhaps `TSRMarker` 😉) that we return from this function. That would also make `clearTSRs` and the `mSimpleFrames` member variable superfluous, since they could be replaced by the RAII object.